Orange/widgets/data/tests/test_owdatasampler.py

```diff
@@ -81,6 +81,42 @@ def test_no_intersection_in_outputs(self):
                     self.assertEqual(len(self.zoo), len(sample) + len(other))
                     self.assertNoIntersection(sample, other)
 
+    def test_bigger_size_with_replacement(self):
+        """Allow bigger output without replacement."""
+        _, out_size, actual_size = self._set_bigger_sample_size(True)
+        self.assertEqual(out_size, actual_size,
+                         'Sample size should not be lowered with replacement')
+
+    def test_bigger_size_without_replacement(self):
+        """Lower output samples to match input's without replacement."""
+        in_size, _, actual_size = self._set_bigger_sample_size(False)
+        self.assertEqual(in_size, actual_size)
+
+    def test_bigger_output_warning(self):
+        self._set_bigger_sample_size(with_replacement=True)
+        self.assertTrue(self.widget.Warning.bigger_sample.is_shown())
+
+    def _set_bigger_sample_size(self, with_replacement):
```

Comment:
I would use a bit different helper function. Something along the lines of:
```python
def set_fixed_sample_size(self, sample_size, with_replacement=False):
    self.select_sampling_type(self.widget.FixedSize)
    self.widget.sampleSizeSpin.setValue(sample_size)
    self.widget.controls.replacement.setChecked(with_replacement)
    self.widget.commit()
```

While this introduces a bit more code duplication in the actual tests, it allows us to reuse the helper in other (future) tests. The same way you were able to use the `select_sampling_type` we wrote for one of our previous tests.